web/js/codeworld.js

```diff
@@ -762,24 +727,6 @@ function help() {
   });
 }
 
-function isEditorClean() {
-  if (!window.codeworldEditor) {
-    return true;
-  }
-
-  const doc = window.codeworldEditor.getDoc();
-
-  if (window.savedGeneration === null) return doc.getValue() === '';
-  else return doc.isClean(window.savedGeneration);
-}
-
-function loadSample(code) {
```

Comment:
Thanks for pointing that out, I overlooked it.

`window.postMessage` doesn't allow sending functions, unless you stringify them and use `eval` on the other side. Not sure this is desirable?

Reply:
Why can't you just replace `parent.loadSample(theCode)` with `parent.postMessage({ type: 'loadSample', code: theCode }, '*')`?  Then instead of defining `loadSample` in funblocks.js and codeworld.js, you'd need to add a case to [the message event handler](https://github.com/google/codeworld/blob/30ff2f7a0a0564ef295121d593362331d863c257/web/js/codeworld.js#L888)

Reply:
I haven't noticed this piece of code before. Thanks for referencing it, that would definitely work!